TF_Service_dotNet/TouchFree/Connections/ClientConnection.cs

```diff
@@ -62,7 +62,24 @@ public void SendHandData(HandFrame _data)
                 return;
             }
 
-            SendResponse(_data, ActionCode.HAND_DATA);
+            CommunicationWrapper<HandFrame> message =
+                new CommunicationWrapper<HandFrame>(ActionCode.HAND_DATA.ToString(), _data);
+
+            string jsonMessage = JsonConvert.SerializeObject(message);
+
+            byte[] jsonAsBytes = Encoding.UTF8.GetBytes(jsonMessage);
+
+            Int32 dataLength = lastHandData.Count;
+
+            IEnumerable<byte> dataToSend = BitConverter.GetBytes(dataLength);
+            dataToSend = dataLength > 0 ? dataToSend.Concat(lastHandData) : dataToSend;
+            dataToSend = dataToSend.Concat(jsonAsBytes);
```

Comment:
If I'm reading this right, `dataToSend` gets transmitted as one blob down the websocket, which the Tooling distinguishes by it being in the form of byte data rather than string data. In that case, do we still benefit from the `CommunicationWrapper`? That JSON will never be decoded by the big switch in `MessageReciever` now

Reply:
Yeah, that would be a good change. Another thing I'd like to add to this message is having the first byte(s) be an identifier for the message type/format to allow it to be changeable or allow different formats of binary messages in future without breaking

Reply:
I've removed the wrapper now and added a flag for the binary data type